source/PlayerInfo.cpp

```diff
@@ -1728,6 +1735,49 @@ bool PlayerInfo::HasTravelPlan() const
 
 
 
+// Verify the player's travel plan and destination are still valid, in case the flagship was changed and/or
+// any needed outfits were removed.
+bool PlayerInfo::HasInvalidTravelPlan() const
+{
+	if(travelDestination && !travelDestination->IsAccessible(flagship.get()))
+		return true;
+	if(!travelPlan.empty())
```

Comment:
I don't think that correctly detects when the player has two routes to a location, but the faster (e.g. automatic and most likely current) route is through restricted space. Example: Going from Zosma[Memory] to Cardea, as long as you have a Jump Drive, DistanceMap knows you can (eventually) get there, but if you had a Quantum Keystone when you made the route, your current map is set to use the wormhole in Terminus.

I wrote the inspection just in case the player had defined some custom route (Shift clicking), as that also handles the case where the player sets some custom route that is different that the automatic route.

For the Jump Drive -> Hyperdrive swap, the "you can't make this jump" error is already raised, but only *after* the player jumps through however many preceeding jumps along hyperspace lanes. HasInvalidTravelPlan will catch that immediately (and !DistanceMap.HasRoute may fail to catch it if there are hyperspace lanes along some other path).